### PR TITLE
Removes RationalFromDoubleException. Replaces this with an explicit M…

### PR DESCRIPTION
**Pull request by timothy-king**
…aybe<Rational> datatype. Makes Maybe<T> CVC4_PUBLIC. Updates the users of Rational::fromDouble(). Miscellaneous cleanup of ApproxSimplex.

### CODE REVIEW

File: src/theory/arith/approx_simplex.cpp [hunk 1]
```diff
@@ -2612,11 +2520,16 @@ bool ApproxGLPK::loadRowSumIntoAgg(int nid, int M, const PrimitiveVec& row_sum){
...
+      return true;
+    }
+
     if(lhs.isKey(x)){
-      lhs.get(x) -= c;
+      lhs.get(x) -= c.value();
     }else{
-      lhs.set(x, -c);
+      lhs.set(x, -c.value());
     }
```
Comment (aniemetz) [inline src/theory/arith/approx_simplex.cpp]: Maybe clang-format the whole if-else block?

File: src/theory/arith/pseudoboolean_proc.cpp [hunk 2]
```diff
@@ -65,17 +66,18 @@ bool PseudoBooleanProcessor::decomposeAssertion(Node assertion, bool negated){
...
 
-    if(d_off.constValue().isIntegral()){
-      d_off = d_off.constValue() + Rational(1) ;
+    if (d_off.value().isIntegral())
+    {
+      d_off = d_off.value() + Rational(1);
     }else{
-      d_off = Rational(d_off.constValue().ceiling());
+      d_off = Rational(d_off.value().ceiling());
     }
```
Comment (aniemetz) [inline src/theory/arith/pseudoboolean_proc.cpp]: Ditto

File: src/theory/arith/simplex_update.h [hunk 3]
```diff
@@ -228,15 +228,16 @@ class UpdateInfo {
...
+  inline int errorsChange() const { return d_errorsChange.value(); }
 
   /**
    * If errorsChange has been set, return errorsChange().
    * Otherwise, return def.
    */
   inline int errorsChangeSafe(int def) const {
-    if(d_errorsChange.just()){
-      return d_errorsChange;
+    if (d_errorsChange)
```
Comment (aniemetz) [inline src/theory/arith/simplex_update.h]: Ditto

File: src/theory/arith/approx_simplex.cpp (hunk 1 above)
Comment (timothy-king) [reply src/theory/arith/approx_simplex.cpp]: Done

File: src/theory/arith/pseudoboolean_proc.cpp (hunk 2 above)
Comment (timothy-king) [reply src/theory/arith/pseudoboolean_proc.cpp]: Done

File: src/theory/arith/simplex_update.h (hunk 3 above)
Comment (timothy-king) [reply src/theory/arith/simplex_update.h]: Done

**Comment by timothy-king** [comment]
PTAL

**Pull request merged**